boltz/api.go

```diff
@@ -451,12 +552,65 @@ func (boltz *Boltz) CreateReverseSwap(request CreateReverseSwapRequest) (*Create
 	return &response, err
 }
 
-func (boltz *Boltz) ClaimReverseSwap(request ClaimReverseSwapRequest) (*PartialSignature, error) {
+func (boltz *Boltz) ClaimReverseSwap(id string, request *ClaimRequest) (*PartialSignature, error) {
+	if boltz.DisablePartialSignatures {
+		return nil, ErrPartialSignaturesDisabled
+	}
+	var response PartialSignature
+	err := boltz.sendPostRequest(fmt.Sprintf("/v2/swap/reverse/%s/claim", id), request, &response)
+
+	if response.Error != "" {
+		return nil, Error(errors.New(response.Error))
+	}
+
+	return &response, err
+}
+
+func (boltz *Boltz) CreateChainSwap(request ChainRequest) (*ChainResponse, error) {
+	var response ChainResponse
+	err := boltz.sendPostRequest("/v2/swap/chain", request, &response)
+
+	if response.Error != "" {
+		return nil, Error(errors.New(response.Error))
+	}
+
+	return &response, err
+}
+
+func (boltz *Boltz) GetChainSwapClaimDetails(swapId string) (*ChainSwapSigningDetails, error) {
+	if boltz.DisablePartialSignatures {
+		return nil, ErrPartialSignaturesDisabled
+	}
+	var response ChainSwapSigningDetails
+	err := boltz.sendGetRequest(fmt.Sprintf("/v2/swap/chain/%s/claim", swapId), &response)
+
+	if response.Error != "" {
+		return nil, Error(errors.New(response.Error))
+	}
+
+	return &response, err
+}
+
+func (boltz *Boltz) SendChainSwapClaimSignature(swapId string, request *ChainSwapSigningRequest) (*PartialSignature, error) {
```

Comment:
Not a fan of the naming. You don't just send it but also get one in response. `ExchangeChainSwapClaimSignature`?